### DISCUSSION TITLE
No user in ssh ?

### DISCUSSION THREAD

**Discussion by thierryforb** [ltsp/ltsp#713] [Q&A]
Hello, 

On previous version of LTSP, I was abble to see how many user were logged in, with the command who for example. I'm running this version of ltsp : 
LTSP version: 22.01-1~ubuntu20.04.1
SERVER XSESSIONS: gnome.desktop  gnome-flashback-compiz.desktop  gnome-flashback-metacity.desktop  gnome-xorg.desktop  ubuntu.desktop  unity.desktop
SERVER OS: PRETTY_NAME="Ubuntu 20.04.4 LTS"
CHROOTS:
VMs:
IMAGES:
x86_64

How can I  see the users logged in now ? 

On a other hand, I would like to survey the ltsp processes with nagios, is there a way to do that ?

Thanks in advance
Thierry

**Comment by alkisg** [comment]
The new LTSP is "lighter", it doesn't create a login (SSH) process for each user, it only creates an SSHFS = SFTP connection. So `who` won't work (unless you modify the SSH config).
You can run `ps` and grep the output of `sshd` and see the users from there.

Regarding nagios, no idea, let's see if anyone else uses it.
Personally I'd also like to check out netdata at some point...

**Reply by thierryforb** [reply]
when I do 
ps faux | grep sshd, i see only my ssh connection to the server.
At the same time, epoptes shows 8 clients running and working

How is it possible ?

**Comment by alkisg** [comment]
Are you using NFS home?

**Reply by thierryforb** [reply]
Not sure I understand your questions, but yes I export the homes via a nfs share to be abble to have shared directories

**Comment by alkisg** [comment]
By default, LTSP uses SSHFS, thus you'd be able to see the user connections with `ps faux | grep sshd`.
If you've manually switched to mounting home over NFS, then you cannot see any user processes on the server as then the users don't even need an account on the server to access the shares.

**Reply by thierryforb** [reply]
How can i be sure ? 
I use this in ltsp.conf : FSTAB_HOME="server:/home /home nfs defaults,nolock 0 0"

**Comment by alkisg** [comment]
Yes, this line means you're using NFS, so there are no user SSH connections or processes running in your server.

**Reply by thierryforb** [reply]
Ok, the is there another way to share directories between user ? use samba shares ? 
Ok, do you know how epoptes can see the current users ?

**Comment by alkisg** [comment]
Samba shares also won't make your users available with commands like `who` on the server. If you do want to have them available with `who`, then use `ltsp remoteapps sleep 1000d` to force a running process on the server. You can add it e.g. in /etc/xdg/autostart, you can google about "how to run a program on login" for details, that part is not LTSP-related.

In epoptes we have a user process running because we need it for thumbnails etc. No such process is required for sharing /home/username, it's not a good idea to enforce one.

**Reply by thierryforb** [reply]
Ok my idea, was perhaps stop using nfs share and replace with samba shares... if I comment the line FSTAB_HOME="server:/home /home nfs defaults,nolock 0 0" in my ltsp.conf the the users will use sshfs and I will be able to see them in the process of my server right ?
And if I stop using nfs shares, then I another solution for shared directories between users, perhaps samba share ... I will google, if there is a way to do such things with sshfs 
Ok If I understand your proposition, adding this : ltsp remoteapps sleep 1000d will start a ssh connection from the client to the serveur and make this connection sleep for 1000 days. This command can be placed in places in ltsp.conf in client part no ?
